torchvision/datasets/_optical_flow.py

```diff
@@ -0,0 +1,149 @@
+import os
+from abc import ABC, abstractmethod
+from glob import glob
+from pathlib import Path
+
+import numpy as np
+import torch
+from PIL import Image
+
+from ..io.image import _read_png_16
+from .vision import VisionDataset
+
+
+__all__ = (
+    "KittiFlow",
+    "Sintel",
+)
+
+
+class FlowDataset(ABC, VisionDataset):
+    def __init__(self, root, transforms=None):
+
+        super().__init__(root=root)
+        self.transforms = transforms
+
+        self._flow_list = []
+        self._image_list = []
+
+    def _read_img(self, file_name):
+        return Image.open(file_name)
+
+    @abstractmethod
+    def _read_flow(self, file_name):
+        # Return the flow or a tuple (flow, valid) for datasets where the valid mask is built-in
+        pass
+
+    def __getitem__(self, index):
+        # Some datasets like Kitti have a built-in valid mask, indicating which flow values are valid
+        # For those we return (img1, img2, flow, valid), and for the rest we return (img1, img2, flow),
+        # and it's up to whatever consumes the dataset to decide what `valid` should be.
+
+        img1 = self._read_img(self._image_list[index][0])
+        img2 = self._read_img(self._image_list[index][1])
+        flow = self._read_flow(self._flow_list[index]) if self._flow_list else None
```

Comment:
Some datasets have a built-in "test" part but this part doesn't contain any target flow value, so we set the flow to `None` in this case.

Reply:
This is similar to `ImageNet`. Worth checking that the policy is aligned.

Reply:
`ImageNet` does not provide a test split. I can't recall which dataset, but I think we agreed that if an piece of information is not available for a specific configuration, we simply return it as `None`.

Reply:
@pmeier ImageNet has a test split but it doesn't have labels. I can't link to it cause you need to register to get it but it's 100k images in a flat folder structure. Here are the 10 first images:
```
$ ls -lh ./imagenet_full_size/061417/test/ | head
total 15G
-rw-r--r-- 1 root root   34K Sep 22  2014 ILSVRC2012_test_00000001.JPEG
-rw-r--r-- 1 root root  120K Sep 22  2014 ILSVRC2012_test_00000002.JPEG
-rw-r--r-- 1 root root   27K Sep 22  2014 ILSVRC2012_test_00000003.JPEG
-rw-r--r-- 1 root root  122K Sep 22  2014 ILSVRC2012_test_00000004.JPEG
-rw-r--r-- 1 root root   97K Sep 22  2014 ILSVRC2012_test_00000005.JPEG
-rw-r--r-- 1 root root  207K Sep 22  2014 ILSVRC2012_test_00000006.JPEG
-rw-r--r-- 1 root root  215K Sep 22  2014 ILSVRC2012_test_00000007.JPEG
-rw-r--r-- 1 root root  178K Sep 22  2014 ILSVRC2012_test_00000008.JPEG
-rw-r--r-- 1 root root   11K Sep 22  2014 ILSVRC2012_test_00000009.JPEG
```

Reply:
Sorry, let me rephrase: our builtin `ImageNet` dataset does not provide a test split:

https://github.com/pytorch/vision/blob/3300692c6e7c2023d2f2356a69ec22ca91e38790/torchvision/datasets/imagenet.py#L22-L28